rundeckapp/grails-spa/packages/ui-trellis/src/library/components/widgets/rundeck-info/RundeckInfo.spec.ts

```diff
@@ -0,0 +1,110 @@
+      ...props,
+    },
+  });
+};
+describe("RundeckInfo", () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("renders RundeckLogo when appInfo.title is 'Rundeck'", async () => {
+    const wrapper = await mountRundeckInfo({});
+    expect(wrapper.findComponent({ name: "RundeckLogo" }).exists()).toBe(true);
+    expect(wrapper.findComponent({ name: "PagerdutyLogo" }).exists()).toBe(
+      false,
+    );
+  });
+  it("renders PagerdutyLogo when appInfo.title is 'Pagerduty'", async () => {
+    const wrapper = await mountRundeckInfo({ appInfo: { title: "Pagerduty" } });
+
+    expect(wrapper.findComponent({ name: "RundeckLogo" }).exists()).toBe(false);
+    expect(wrapper.findComponent({ name: "PagerdutyLogo" }).exists()).toBe(
+      true,
+    );
+  });
+  it("renders RundeckVersions information correct props based on version data", async () => {
+    const versionProps = {
+      logo: false,
+      logocss: "some-css",
+      number: "1.0.0",
+      tag: "stable",
+      title: "Rundeck",
+    };
+    const wrapper = await mountRundeckInfo({ version: versionProps });
+
+    await wrapper.vm.$nextTick();
+    const versionComponent = wrapper.findComponent({ name: "RundeckVersion" });
+    expect(versionComponent.exists()).toBe(true);
+    expect(versionComponent.props()).toEqual(versionProps);
+  });
+  it("renders ServerDisplay correctly based on a server props", async () => {
+    const serverProps = {
+      name: "localHost",
+      icon: "paperclip",
+      uuid: "uuid1",
+      showId: true,
+    };
+    const wrapper = await mountRundeckInfo({ server: serverProps });
+    await wrapper.vm.$nextTick();
+    const serverComponent = wrapper.findComponent({ name: "ServerDisplay" });
+    expect(serverComponent.exists()).toBe(true);
+    expect(serverComponent.props()).toEqual({
+      name: serverProps.name,
+      glyphicon: serverProps.icon,
+      uuid: serverProps.uuid,
+      showId: serverProps.showId,
+    });
+  });
+
+  it("renders version display correctly", async () => {
+    const versionDisplay = {
+      name: "Rundeck",
+      icon: "icon-url",
+      color: "lamp",
+    };
+    const wrapper = await mountRundeckInfo({ version: versionDisplay });
+    const versionDisplayComponent = wrapper.findComponent({
+      name: "VersionDisplay",
+    });
+
+    expect(versionDisplayComponent.props("text")).toBe("Rundeck lamp icon-url");
+  });
+
+  it("renders latest release information correctly ", async () => {
+    const latestProps = {
+      full: "2.0.0",
+      tag: "GA",
+    };
+    const wrapper = await mountRundeckInfo({
+      latest: latestProps,
+    });
+
+    const latestSection = wrapper
+      .find(".rundeck-info-widget__latest")
+      .findComponent({ name: "RundeckVersion" });
+    expect(latestSection.exists()).toBe(true);
+
+    expect(latestSection.props("number")).toEqual(latestProps.full);
+    expect(latestSection.props("tag")).toEqual(latestProps.tag);
```

Comment:
Also, would like you to add another expect here, checking that the span of rundeckVersion has the expected text

Reply:
> Also, would like you to add another expect here, checking that the span of rundeckVersion has the expected text
I have question regarding latest release: 
In the latest release even though tag is included but the output it is giving "Rundeck and number". Then I checked the computed property and it seems like tag is not included  
```
computed: {
    cssclass(): string {
      return `${this.logocss} app-logo`;
    },
    text(): string {
      const { title, number, tag } = this;
      let text = "";
      text += `${title ? title + " " : ""}`;
      text += `${number}`;
      return text;
    },
  },
```
Two ways I am thinking to fix it. Either I include tag in the computed property or don't use tag in the test as the actual output it is asking `{ title: "Rundeck", full: "1.0.0" }`

I am sending fix. Please review and suggest. Thanks

Reply:
> > Also, would like you to add another expect here, checking that the span of rundeckVersion has the expected text
> > I have question regarding latest release:
> > In the latest release even though tag is included but the output it is giving "Rundeck and number". Then I checked the computed property and it seems like tag is not included
> 
> ```
> computed: {
>     cssclass(): string {
>       return `${this.logocss} app-logo`;
>     },
>     text(): string {
>       const { title, number, tag } = this;
>       let text = "";
>       text += `${title ? title + " " : ""}`;
>       text += `${number}`;
>       return text;
>     },
>   },
> ```
> 
> Two ways I am thinking to fix it. Either I include tag in the computed property or don't use tag in the test as the actual output it is asking `{ title: "Rundeck", full: "1.0.0" }`
> 
> I am sending fix. Please review and suggest. Thanks

realized that span of rundeckVersion has the expected text of` title and number` and no need to test tag there. Added tag in my new commit which I removed in this commit due to confusion.